x-pack/plugins/fleet/public/hooks/use_create_cloud_formation_url.ts

```diff
@@ -71,15 +71,18 @@ const createCloudFormationUrl = (
   templateURL: string,
   enrollmentToken: string,
   fleetUrl: string,
-  kibanaVersion: string,
+  agentVersion: string,
   awsAccountType: CloudSecurityIntegrationAwsAccountType | undefined
 ) => {
   let cloudFormationUrl;
 
+  /*
+    template url has `&param_ElasticAgentVersion=KIBANA_VERSION` part. KIBANA_VERSION is used for templating as agent version used to match Kibana version, but now it's not necessarily the case
+   */
   cloudFormationUrl = templateURL
     .replace('FLEET_ENROLLMENT_TOKEN', enrollmentToken)
     .replace('FLEET_URL', fleetUrl)
-    .replace('KIBANA_VERSION', kibanaVersion);
+    .replace('KIBANA_VERSION', agentVersion);
```

Comment:
So just to get it straight, what we actually wanted to pass was the agent version but instead we passed a kibana version only because they used to be the same? at this case this fix was needed regardless. and yes i completely agree the naming here is confusing

Reply:
yep, it was always the agent version really, it just used to match the Kibana version before serverless